### PR TITLE
Fixes, more test cases & coverage, improved slide seed generation, allowed titles to be given as argument, removal of unused imports

### PR DESCRIPTION
**Pull request by twinters**

### CODE REVIEW

**Comment by korymath** [comment]
Big review! 

I will get into this tomorrow. Nice work on this. 30 commits and a huge amount of progress.

File: talkgenerator/sources/phrasefinder.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import requests
+
+from talkgenerator.util import language_util
+
+URL = "https://api.phrasefinder.io/search?corpus=eng-us&query={}&nmax=1"
+
+
+def _search(word):
```
Comment (korymath) [inline talkgenerator/sources/phrasefinder.py:8]: Are there unit tests for these functions?

File: tests/test_phrasefinder.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import unittest
+
+from talkgenerator.sources import phrasefinder
+
+
+class PhraseFinderTest(unittest.TestCase):
```
Comment (korymath) [inline tests/test_phrasefinder.py:6]: Indeed there are tests, nice work.

**Review by korymath — APPROVED** [review]
Fantastic work on the major review and revisions. Nice added functionality as well.

LGTM

File: talkgenerator/utils.py [hunk 3]
```diff
@@ -120,9 +125,9 @@ def str2bool(v):
 
 def get_argument_parser():
     parser = argparse.ArgumentParser(description='Quickly build a slide deck.')
-    parser.add_argument('--topic', default='cat', type=str,
+    parser.add_argument('--topic', default='', type=str,
                         help="Topic of presentation.")
-    parser.add_argument('--num_slides', default=10, type=int,
+    parser.add_argument('--num_slides', '--slides', default=10, type=int,
```
Comment (korymath) [inline talkgenerator/utils.py:130]: @twinters ... is there a reason we use multiple keyword arguments here?

**Review by korymath — COMMENTED** [review]
Multiple keyword args on the run code arguments list?

File: talkgenerator/utils.py (hunk 3 above)
Comment (twinters) [reply talkgenerator/utils.py:130]: I find "slides" to be a more natural name for this parameter, but don't want to break existing installations.

Comment (korymath) [reply talkgenerator/utils.py:130]: Cool!

**Pull request merged**